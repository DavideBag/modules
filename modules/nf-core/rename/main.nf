process RENAME {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/fastx_toolkit:0.0.13--0':
        'biocontainers/fastx_toolkit:0.0.13--0' }"

    input:
    tuple val(meta), path(reads)
    val options

    output:
    tuple val(meta), path("*.fastq.gz"), emit: renamed
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    if (options.single_end) {
        """
        zcat ${reads[0]} | fastx_renamer -z -n COUNT -o ${meta.id}_renamed.fastq.gz
        """
    }
    else {
        """
        zcat ${reads[0]} | fastx_renamer -z -n COUNT -o ${meta.id}_R1_renamed.fastq.gz
        zcat ${reads[1]} | fastx_renamer -z -n COUNT -o ${meta.id}_R2_renamed.fastq.gz


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rename: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """
    }
}
