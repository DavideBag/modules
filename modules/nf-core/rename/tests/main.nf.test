nextflow_process {
    name "Test Process RENAME"
    script "../main.nf"
    process "RENAME"
    config "./nextflow.config"
    tag "modules"
    tag "modules_nfcore"
    tag "rename"
    tag "renamingflash"

    test("rename-single_end") {
        when {
            process {
                """
                input[0] = [ [id:"null"], file("/workspace/modules/modules/nf-core/rename/tests/test_samples.tsv", checkIfExists: true)]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("/workspace/modules/modules/nf-core/rename/tests/test_samples.tsv", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }

}