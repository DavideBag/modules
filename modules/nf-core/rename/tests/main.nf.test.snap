{
    "rename": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "renamed": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-16T09:49:03.81881755"
    }
}